### PR TITLE
initial models commit

### PR DESCRIPTION
**Pull request by manyuagashe**

### CODE REVIEW

**Review by aidankim12 — APPROVED** [review]
Worked together with Manyu as he created the file.

File: backend/models/service_model.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from typing import Optional
+from .enum_for_models import ProgramTypeEnum
+
+
+class Service(BaseModel):
+    id: int | None = None
+    created_at: datetime | None = None
+    name: str
+    status: str
+    summary: str
```
Comment (bhardee16) [inline backend/models/service_model.py:14]: This is good, but I think for consistency sake, we should either stick with defining as `str` or `Field(..., max_length=150, description="The name of the service")`. Similar to how Resource was defined.

File: backend/models/tag_model.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from datetime import datetime
+from typing import Optional
+
+
+class Tag(BaseModel):
+    id: int
+    content: str = Field(
+        ..., max_length=600, description="content associated with the tag"
+    )
+    created_at: datetime | None = None
```
Comment (bhardee16) [inline backend/models/tag_model.py:13]: For consistency: either `created_at: Optional[datetime]` or `created_at: datetime | None = None`.

File: backend/models/service_model.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from enum import Enum
+from typing import List
+from datetime import datetime
+from typing import Optional
+from .enum_for_models import ProgramTypeEnum
+
+
+class Service(BaseModel):
+    id: int | None = None
+    created_at: datetime | None = None
```
Comment (bhardee16) [inline backend/models/service_model.py:11]: For consistency: either created_at: Optional[datetime] or created_at: datetime | None = None.

File: backend/models/resource_model.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from .enum_for_models import ProgramTypeEnum
+
+
+class Resource(BaseModel):
+    id: int | None = None
+    name: str = Field(..., max_length=150, description="The name of the resource")
+    summary: str = Field(..., max_length=300, description="The summary of the resource")
+    link: str = Field(..., max_length=150, description="link to the resource")
+    programtype: ProgramTypeEnum
+    created_at: Optional[datetime]
```
Comment (bhardee16) [inline backend/models/resource_model.py:15]: For consistency: either created_at: Optional[datetime] or created_at: datetime | None = None.

File: backend/models/user_model.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+class User(BaseModel):
+    id: int | None = None
+    username: str = Field(..., description="The username of the user")
+    email: str = Field(..., description="The e-mail of the user")
+    experience: int = Field(..., description="Years of Experience of the User")
+    group: str
+    programtype: List[ProgramTypeEnum]
+    usertype: UserTypeEnum
+    created_at: Optional[datetime]
```
Comment (bhardee16) [inline backend/models/user_model.py:17]: For consistency: either created_at: Optional[datetime] or created_at: datetime | None = None.

**Review by bhardee16 — CHANGES_REQUESTED** [review]

**Review by bhardee16 — APPROVED** [review]

**Pull request merged**